javascript/react/redux-i/use-redux-with-react-2.md

```diff
@@ -0,0 +1,105 @@
+---
+author: kapnobatai136
+
+levels:
+
+  - basic
+
+  - advanced
+
+  - medium
+
+type: normal
+
+category: must-know
+
+
+links:
+
+  - '[redux.js.org](https://redux.js.org/basics/usage-with-react){documentation}'
+
+notes: This can be also placed in the react subtopic
+
+aspects:
+  - introduction
+
+---
+
+# Use Redux with React (2)
+
+---
+## Content
+
+Now, looking in `Counter.js`, we must connect the component to our `store`:
+
+```jsx
+// Counter.js
+import React from "react";
+import { connect } from "react-redux";
+
+class Counter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      count: 0
+    };
+  }
+  render() {
+    // ...
+  }
+}
+
+function mapStateToProps(state) {
+  return {
+    count: state.count
+  };
+}
+
+export default connect(mapStateToProps)(
+  Counter
+);
+```
+
+Let's look at this code step-by-step. We want to connect our `<Counter />` component to our `store`, and to do this we have to make use of the `connect()` method. What this method does is that it takes our component as input, and returns a **new** connected component class that wraps our `<Counter />` component. Notice how you pass the `mapStateToProps` and `Counter` to the `connect` method. This is because `connect()` is a higher order function, which basically means it is a "function that returns a function" and the function `connect` returns, in this case, is one that takes `Counter` as an argument and returns a new `Counter` that is connected to the redux store.
```

Comment:
Just a tid bit more lol

```suggestion
Let's look at this code step-by-step. We want to connect our `<Counter />` component to our `store`, and to do this we have to make use of the `connect()` method. What this method does is that it takes our component as input, and returns a **new** connected component class that wraps our `<Counter />` component. Notice how you pass the `mapStateToProps` and `Counter` to the `connect` method, through separate function calls. This is because `connect()` is a higher order function, which basically means it is a "function that returns a function". The function that `connect` returns, in this case, is one that takes `Counter` as an argument and returns a new `Counter` that is connected to the redux store.
```